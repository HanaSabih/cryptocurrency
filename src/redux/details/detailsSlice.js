const apiCoin = 'https://api.coincap.io/v2/assets/';

const GET_DETAIL = 'GET_DETAIL';
const initialState = [{ id: 1, symbol: 'CrytoCoin' }];

const fetchDetail = async (id) => {
  const get = await fetch(apiCoin + id, { method: 'GET' });
  const coins = await get.json();
  return coins;
};

export const coinDetail = (id) => async (dispatch) => {
  const cDetail = await fetchDetail(id);
  dispatch({
    type: GET_DETAIL,
    detail: cDetail.data,
  });
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_DETAIL:
      return action.detail;
    default:
      return state;
  }
};

export default reducer;
