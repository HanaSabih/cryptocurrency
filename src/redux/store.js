import {
  combineReducers,
} from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import cryto from './coins/coinsSlice';
import detail from './details/detailsSlice';

const reducer = combineReducers({
  cryto,
  detail,
});

const store = configureStore({
  reducer,
});

export default store;
