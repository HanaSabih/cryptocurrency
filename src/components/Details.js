import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import Card from 'react-bootstrap/Card';
import ListGroup from 'react-bootstrap/ListGroup';
import { coinDetail } from '../redux/details/detailsSlice';

const Details = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const cryto = useSelector((store) => store.detail);
  useEffect(() => {
    dispatch(coinDetail(id));
  });
  const {
    symbol, name: title, rank, priceUsd, volumeUsd24Hr, supply,
  } = cryto;
  return (

    <Card className="text-center border-0 ">
      <ListGroup variant="flush">
        <ListGroup.Item variant="flush" className="py-5 d-flex flex-row justify-content-between align-items-center text-white">
          {' '}
          <h2 className="fs-1 ">{symbol}</h2>
          <h3 className="fs-1">{title}</h3>
        </ListGroup.Item>
        <ListGroup.Item className="text-white ">Details</ListGroup.Item>
        <ListGroup.Item className="d-flex flex-row justify-content-between align-items-center py-2 text-white">
          <span>Rank:</span>
          <span>
            #
            {rank}
          </span>
        </ListGroup.Item>
        <ListGroup.Item className="d-flex flex-row justify-content-between align-items-center py-5 text-white">
          <span>
            Price
            {' USD'}
            :
          </span>
          <span>
            {' $'}
            {priceUsd}

          </span>
        </ListGroup.Item>
        <ListGroup.Item className="d-flex flex-row justify-content-between align-items-center py-5 text-white">
          <span>Volume 24h:</span>
          <span>
            {' $'}
            {volumeUsd24Hr}
          </span>
        </ListGroup.Item>
        <ListGroup.Item className="d-flex flex-row justify-content-between align-items-center py-5 text-white">
          <span>Total Supply:</span>
          <span>
            {symbol}
            {' '}
            {supply}

          </span>
        </ListGroup.Item>

      </ListGroup>
    </Card>

  );
};

export default Details;
