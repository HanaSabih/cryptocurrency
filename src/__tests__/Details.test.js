import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Details from '../components/Details';
import { coinDetail } from '../redux/details/detailsSlice';

describe('Details component', () => {
  test('component match with the snapshot', () => {
    store.dispatch(coinDetail('bitcoin'));
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
