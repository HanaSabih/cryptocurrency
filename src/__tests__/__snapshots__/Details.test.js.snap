// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component component match with the snapshot 1`] = `
<div
  className="text-center border-0  card"
>
  <div
    className="list-group list-group-flush"
    onKeyDown={[Function]}
  >
    <div
      className="py-5 d-flex flex-row justify-content-between align-items-center text-white list-group-item list-group-item-flush"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
       
      <h2
        className="fs-1 "
      />
      <h3
        className="fs-1"
      />
    </div>
    <div
      className="text-white  list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      Details
    </div>
    <div
      className="d-flex flex-row justify-content-between align-items-center py-2 text-white list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <span>
        Rank:
      </span>
      <span>
        #
      </span>
    </div>
    <div
      className="d-flex flex-row justify-content-between align-items-center py-5 text-white list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <span>
        Price
         USD
        :
      </span>
      <span>
         $
      </span>
    </div>
    <div
      className="d-flex flex-row justify-content-between align-items-center py-5 text-white list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <span>
        Volume 24h:
      </span>
      <span>
         $
      </span>
    </div>
    <div
      className="d-flex flex-row justify-content-between align-items-center py-5 text-white list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <span>
        Total Supply:
      </span>
      <span>
         
      </span>
    </div>
  </div>
</div>
`;
